### ISSUE TITLE
don't use __defineGetter__ / __defineSetter__

### ISSUE THREAD

**Issue by TitanNano** [loqui/im#632]
why are we using **defineGetter** and **defineSetter**? It's not a standard and probably gets removed in the future. We should use Object.defineProperty().

**Comment by aesedepece** [comment]
Good point, let's move the three **define*etter** to Object.defineProperty().
